dns_forwarder_lb_vmss/main.tf

```
@@ -0,0 +1,190 @@
+  frontend_private_ip_address_lb = var.static_address_lb != null ? var.static_address_lb : cidrhost(var.address_prefixes_lb, 4)
+
+  subnet_vmss_id = var.subnet_vmss_id != null ? var.subnet_vmss_id : module.subnet_vmss[0].id
+  subnet_lb_id   = var.subnet_lb_id != null ? var.subnet_lb_id : module.subnet_lb[0].id
+
+}
+
+#
+# Subnet virtual machine scale set
+#
+
+module "subnet_vmss" {
+  source = "git::https://github.com/pagopa/terraform-azurerm-v3.git//subnet?ref=v7.47.0"
+  count  = var.subnet_vmss_id != null ? 0 : 1
+
+  name                 = "${local.prefix}-vmss-snet"
+  resource_group_name  = var.resource_group_name
+  virtual_network_name = var.virtual_network_name
+  address_prefixes     = [var.address_prefixes_vmss]
+}
+
+#
+# Subnet load balancer
+#
+
+module "subnet_lb" {
+  source = "git::https://github.com/pagopa/terraform-azurerm-v3.git//subnet?ref=v7.47.0"
+  count  = var.subnet_lb_id != null ? 0 : 1
+
+  name                 = "${local.prefix}-lb-snet"
+  resource_group_name  = var.resource_group_name
+  virtual_network_name = var.virtual_network_name
+  address_prefixes     = [var.address_prefixes_lb]
+}
+
+#
+# Network security group - virtual machine scale set
+#
+
+resource "azurerm_network_security_group" "vmss" {
+  count = var.subnet_vmss_id != null ? 0 : 1
+
+  name                = "${local.prefix}-vmss-nsg"
+  location            = var.location
+  resource_group_name = var.resource_group_name
+
+  # Inbound rule
+  security_rule {
+    name                       = "${local.prefix}-vmss-dns-tcp-rule"
+    priority                   = 200
+    direction                  = "Inbound"
+    access                     = "Allow"
+    protocol                   = "Tcp"
+    source_port_range          = "53"
+    destination_port_range     = "53"
+    source_address_prefix      = "VirtualNetwork"
+    destination_address_prefix = "168.63.129.16" # https://learn.microsoft.com/en-us/azure/virtual-network/what-is-ip-address-168-63-129-16
+  }
+
+  security_rule {
+    name                       = "${local.prefix}-vmss-dns-udp-rule"
+    priority                   = 210
+    direction                  = "Inbound"
+    access                     = "Allow"
+    protocol                   = "Udp"
+    source_port_range          = "53"
+    destination_port_range     = "53"
+    source_address_prefix      = "VirtualNetwork"
+    destination_address_prefix = "168.63.129.16" # https://learn.microsoft.com/en-us/azure/virtual-network/what-is-ip-address-168-63-129-16
+  }
+
+  security_rule {
+    name                       = "${local.prefix}-vmss-deny-all-rule"
+    priority                   = 4096
+    direction                  = "Inbound"
+    access                     = "Deny"
+    protocol                   = "*"
+    source_port_range          = "*"
+    destination_port_range     = "*"
+    source_address_prefix      = "*"
+    destination_address_prefix = "*"
+  }
+}
+
+#
+# Load balancer
+#
+
+module "lb" {
```

Please can you change the name in


```suggestion
module "load_balancer" {
```